packages/react-native-gesture-handler/src/v3/HostGestureDetector.web.tsx

```diff
@@ -0,0 +1,68 @@
+import { useEffect, useRef, useCallback } from 'react';
+import { View } from 'react-native';
+import RNGestureHandlerModule from '../RNGestureHandlerModule.web';
+import { ActionType } from '../ActionType';
+import { PropsRef } from '../web/interfaces';
+
+export interface GestureHandlerDetectorProps extends PropsRef {
+  handlerTags: number[];
+  dispatchesAnimatedEvents: boolean;
+  moduleId: number;
+  children?: React.ReactNode;
+}
+
+const HostGestureDetector = (props: GestureHandlerDetectorProps) => {
+  const { handlerTags, dispatchesAnimatedEvents, children } = props;
+
+  const viewRef = useRef(null);
+  const propsRef = useRef<PropsRef>(props);
+  const attachedHandlerTags = useRef<Set<number>>(new Set<number>());
+
+  const detachHandlers = useCallback((oldHandlerTags: Set<number>) => {
+    oldHandlerTags.forEach((tag) => {
+      RNGestureHandlerModule.detachGestureHandler(tag);
+    });
+  }, []);
+
+  const attachHandlers = useCallback(
+    (currentHandlerTags: Set<number>) => {
+      const oldHandlerTags =
+        attachedHandlerTags.current.difference(currentHandlerTags);
+      const newHandlerTags = currentHandlerTags.difference(
+        attachedHandlerTags.current
+      );
+
+      detachHandlers(oldHandlerTags);
+
+      newHandlerTags.forEach((tag) => {
+        RNGestureHandlerModule.attachGestureHandler(
+          tag,
+          viewRef.current,
+          ActionType.NATIVE_DETECTOR,
+          propsRef,
+          dispatchesAnimatedEvents
+        );
+      });
+      attachedHandlerTags.current = currentHandlerTags;
+    },
+    [dispatchesAnimatedEvents]
+  );
+
+  useEffect(() => {
+    attachHandlers(new Set(handlerTags));
+  }, [attachHandlers]);
```

Comment:
```suggestion
  }, [handlerTags]);
```

And I think this should have `handlerTags` in dependencies to make sure it re-runs when they change.

Reply:
Thanks, it got lost during memoisation. Fixed [2443971](https://github.com/software-mansion/react-native-gesture-handler/pull/3637/commits/244397112c067beb6fd5fce79c2f706e1ddfe850).